### PR TITLE
Adding a SYMFONY_DOCKER env var to permanently activate docker support

### PR DESCRIPTION
**Pull request by weaverryan**
The partner PR to: https://github.com/symfony-cli/symfony-cli/pull/53

Will allow for a `symfony new --docker` flag, which will start a project with the Docker configuration pre-activated.

### CODE REVIEW

File: src/Configurator/DockerComposeConfigurator.php [hunk 1]
```diff
@@ -111,34 +111,12 @@ public static function shouldConfigureDockerRecipe(Composer $composer, IOInterfa
...
-            $question,
-            function ($value) {
-                if (null === $value) {
-                    return 'y';
-                }
-                $value = strtolower($value[0]);
-                if (!\in_array($value, ['y', 'n', 'p', 'x'], true)) {
-                    throw new \InvalidArgumentException('Invalid choice.');
-                }
+        if ($_SERVER['DOCKER_SUPPORT'] ?? false) {
```
Comment (nicolas-grekas) [inline src/Configurator/DockerComposeConfigurator.php]: let's use `filter_var(..., \FILTER_VALIDATE_BOOLEAN)` and allow enabling or disabling via the env var?

Comment (nicolas-grekas) [reply src/Configurator/DockerComposeConfigurator.php]: what about SYMFONY_DOCKER for the name?

Comment (fabpot) [reply src/Configurator/DockerComposeConfigurator.php]: Re: enabling/disabling, we want Docker disables by default, and enable it when the env var is "set". We don't really care about the value. If it's false-ish, it's like not having it.

Comment (weaverryan) [reply src/Configurator/DockerComposeConfigurator.php]: Name updated.

> Re: enabling/disabling, we want Docker disables by default, and enable it when the env var is "set". We don't really care about the value. If it's false-ish, it's like not having it.

And so `SYMFONY_DOCKER=0` would work if you, for some reason, wanted to use this env var but explicitly set it to false.

Comment (nicolas-grekas) [reply src/Configurator/DockerComposeConfigurator.php]: it's strange to be able to SYMFONY_DOCKER=0 to enable it

It's easy cheap to implement better now isn't it?

Comment (fabpot) [reply src/Configurator/DockerComposeConfigurator.php]: SYMFONY_DOCKER=0 must disable it, that's what I mentioned before.

Comment (nicolas-grekas) [reply src/Configurator/DockerComposeConfigurator.php]: oh, right about SYMFONY_DOCKER=0
my point stands for SYMFONY_DOCKER=false :)
using `filter_var(..., \FILTER_VALIDATE_BOOLEAN)` is the way to check boolean envvars (or php.ini settings) in Symfony to me

Comment (dunglas) [reply src/Configurator/DockerComposeConfigurator.php]: So `SYMFONY_DOCKER=1` to permanently enable Docker support and `SYMFONY_DOCKER=0` to permanently disable it (without being asked)?

+1 to use `filter_var()`.

**Comment by fabpot** [comment]
Thank you @weaverryan.

**Pull request merged**